### PR TITLE
use defaultChannelSlug in queries and mutations

### PR DESCRIPTION
**Pull request by AlicjaSzu**
I want to merge this change because...

- Store the channel slug in the storefront 
- Update queries that require channel slug and pass the value to these queries to get channel’s data.
- Update featured products - use products from channel with featured-products slug

### Pull Request Checklist

1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [x] Changes are mentioned in the changelog.

### Test Environment Config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-458 Use Channel SLUG in storefront](https://app.clickup.com/t/2549495/SALEOR-458)

**Review by dominik-zeglen — APPROVED** [review]

**Review by orzechdev — APPROVED** [review]

**Comment by jwm0** [comment]
⚠️ This PR uses temporary non-released SDK version that should not be merged into master - commit 1fb1a01

**Comment by krzysztofwolski** [comment]
Waiting for stage server update, then tests should pass

**Comment by patrys** [comment]
Task linked: [SALEOR-1645 Drop deprecated navigation field from shop query](https://app.clickup.com/t/2549495/SALEOR-1645)

**Review by jwm0 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -177,6 +177,10 @@ npm run generate
     - Check to see that "Less secure app access" is turned ON. Under "Manage your Google Account" > Go to the security tab. By default, the setting is off for security reasons.
     - If using 2FA make sure to set an [app password](https://support.google.com/accounts/answer/185833?p=InvalidSecondFactor&visit_id=637355441414497566-1310044707&rd=1) and use that in place of your normal login password.
 
+### Multichannel
+
+- **Set [DEFAULT_CHANNEL_SLUG] environment variable.**
```
Comment (patrys) [inline README.md:182]: Why do we call it the "default channel slug"? Are there non-default slugs used by the Storefront?

Comment (krzysztofwolski) [reply README.md:182]: @patrys No, currently there is only one.

Comment (krzysztofwolski) [reply README.md]: @patrys changed

**Review by patrys — APPROVED** [review]

**Pull request merged**